src/xdg-autostart-generator/xdg-autostart-service.c

```diff
@@ -0,0 +1,655 @@
+                        copy = NULL;
+
+                        start = end + 1;
+                }
+        }
+
+        /* Any trailing entry should be ignored if it is empty. */
+        if (end > start) {
+                r = strv_extend (sv, start);
+                if (r < 0)
+                        return r;
+        }
+
+        return 0;
+}
+
+static int config_item_xdg_lookup(
+                const void *table,
+                const char *section,
+                const char *lvalue,
+                ConfigParserCallback *func,
+                int *ltype,
+                void **data,
+                void *userdata)
+{
+        XdgAutostartService *service = userdata;
+        _cleanup_free_ char *lvalue_stripped = NULL;
+
+        assert(table == NULL);
+        assert(lvalue);
+        assert(func);
+        assert(ltype);
+        assert(data);
+
+        if (!streq_ptr(section, "Desktop Entry"))
+                return 0;
+
+        /* Space before and after the = sign should be ignored. */
+        lvalue_stripped = delete_trailing_chars(strdup(lvalue), NULL);
+
+        /* We don't need to handle any localised fields. */
+        if (streq(lvalue_stripped, "Name")) {
+                *func = xdg_config_parse_string;
+                *data = &service->description;
+        } else if (streq(lvalue_stripped, "Exec")) {
+                *func = xdg_config_parse_string;
+                *data = &service->exec_string;
+        } else if (streq(lvalue_stripped, "TryExec")) {
+                *func = xdg_config_parse_string;
+                *data = &service->try_exec;
+        } else if (streq(lvalue_stripped, "Type")) {
+                *func = xdg_config_parse_string;
+                *data = &service->type;
+        } else if (streq(lvalue_stripped, "OnlyShowIn")) {
+                *func = xdg_config_parse_strv;
+                *data = &service->only_show_in;
+        } else if (streq(lvalue_stripped, "NotShowIn")) {
+                *func = xdg_config_parse_strv;
+                *data = &service->not_show_in;
+        } else if (streq(lvalue_stripped, "Hidden")) {
+                *func = xdg_config_parse_bool;
+                *data = &service->hidden;
+        } else if (streq(lvalue_stripped, "AutostartCondition")) {
+                *func = xdg_config_parse_string;
+                *data = &service->autostart_condition;
+        } else if (streq(lvalue_stripped, "X-KDE-autostart-condition")) {
+                *func = xdg_config_parse_string;
+                *data = &service->kde_autostart_condition;
+        } else if (streq(lvalue_stripped, "X-GNOME-Autostart-Phase")) {
+                *func = xdg_config_parse_string;
+                *data = &service->gnome_autostart_phase;
+        } else if (streq(lvalue_stripped, "X-GNOME-HiddenUnderSystemd")) {
+                /* FIXME: X-systemd-skip? */
+                *func = xdg_config_parse_bool;
+                *data = &service->systemd_skip;
+        }
```

Comment:
use a table for this? i.e. an aray of structures?

Reply:
i.e. i think this should really use the generic config_item_table_lookup() call we already have, and then define the functions in a ConfigTableItem array

Reply:
Hah, yeah, we can chain up to `config_item_table_lookup`.

I think I only realised that `config_item_table_lookup` doesn't work as is because we need to strip whitespace in front of the `=` sign. But it should work perfectly fine to pass on `lvalue_stripped`.